.buildkite/dagster-buildkite/dagster_buildkite/git.py

```diff
@@ -0,0 +1,40 @@
+import logging
+import os
+import subprocess
+from pathlib import Path
+from typing import Set
+
+
+def get_commit(rev):
+    return subprocess.check_output(["git", "rev-parse", "--short", rev]).decode("utf-8").strip()
+
+
+class ChangedFiles:
+    _repositories: Set[Path] = set()
+    all: Set[Path] = set()
+
+    @classmethod
+    def load_repository(cls, git_repository_directory: Path) -> None:
+        # Only do the expensive git diffing once
+        if git_repository_directory in cls._repositories:
+            return None
+
+        original_directory = os.getcwd()
+        os.chdir(git_repository_directory)
+
+        # TODO: Make it possible to compare to BUILDKITE_PULL_REQUEST_BASE_BRANCH
```

Comment:
This also needs to be responsive to `BUILDKITE_COMMIT` and `BUILDKITE_BRANCH` - maybe a few other things too...